Core/Src/bms_can_monitor.c

```diff
@@ -0,0 +1,51 @@
+#include "bms_can_monitor.h"
+#include "timer.h"
+#include "fault.h"
+#include "can.h"
+#include "cerberus_conf.h"
+#include <assert.h>
+#include <stdlib.h>
+
+#define BMS_WATCHDOG_DURATION 10 /*Duration of btween petting bms monitor watchdog*/
+#define CAN_QUEUE_SIZE 5
+
+osThreadId_t bms_can_monitor_handle;
+const osThreadAttr_t bms_can_monitor_attributes = {
+	.name = "BMSCANMonitor",
+	.stack_size = 128 * 8,
+	.priority = (osPriority_t)osPriorityLow1 /*Adjust priority*/
+};
+
+bms_can_monitor_t* bms_can_monitor_init() {
+
+    bms_can_monitor_t* bms_can_monitor = malloc(sizeof(bms_can_monitor_t));
+	assert(bms_can_monitor);
+
+    bms_can_monitor_queue = osMessageQueueNew(CAN_QUEUE_SIZE, sizeof(can_msg_t), NULL);
+    assert(bms_can_monitor_queue);
+
+    return bms_can_monitor;
+}
+
+void vBMSCANMonitor(void* pv_params) 
+{
+	fault_data_t fault_data = { .id = BMS_CAN_MONITOR_FAULT, .severity = DEFCON1 }; /*Ask about severity*/
+    
+    bms_can_monitor_t* bms_can_monitor = (bms_can_monitor_t*)pv_params;
+	can_msg_t msg_from_queue;
+
+	for (;;) {
+		if (osOK == osMessageQueueGet(bms_can_monitor_queue, &msg_from_queue, NULL, osWaitForever)) {
+			if (!is_timer_active(&bms_can_monitor->timer)) {
+				start_timer(&bms_can_monitor->timer, BMS_WATCHDOG_DURATION);
+			} else {
+				cancel_timer(&bms_can_monitor->timer);
```

Comment:
Why are we canceling the timer? We shouldn't eve cancel the timer in this since it should just run out. We are gonna want to continuously restart the timer tho whenever this is called. Tbh an RTOS timer is probably good for this purpose I think. Read up on the timer implementation here: https://www.keil.com/pack/doc/CMSIS/RTOS2/html/group__CMSIS__RTOS__TimerMgmt.html.

With the new task structure, I think what we are going to want to continuously restart a callback which will get triggered when the timer runs out (we don't get CAN message). This function will then queue the fault.